### DISCUSSION TITLE
Q: Does --resync respect skip_dir?

### DISCUSSION THREAD

**Discussion by wanageeska** [abraunegg/onedrive#2184] [Q&A] [answered]
Hi,

I lost my hdd recently (recovered most of it) and am in the process of resyncing my onedrive directory. I have a very large directory online in OneDrive (~500GB) that I don't want downloaded when I `--resync` because I'm currently travelling and on mobile data.

I came across the comment below in #1968:
> Files are not re-downloaded. The online state and integrity of the files locally is checked. If the file is already present locally, it is not downloaded, if it is missing, it is downloaded in accordance with your desired configuration.

Am I correct in thinking that if I add this large directory ('LIBRARY') to `skip_dir` in my config then it won't be downloaded when I `--resync`?

The reason I'm asking is because I tested this new config by running a `--resync --dry-run` and the terminal output indicates that the client *is* downloading the files in this directory. (Screenshot of output attached.)

On seeing that output, I'm now obviously a little hesitant to pull the trigger on my `--resync`. When the client says that it's *' Downloading file...'*  does it mean the actual file or just a 'record' of the file's existence? Is anyone able to help me clarify this before I go doing something regrettable?

Thanks in advance,
-w

![image](https://user-images.githubusercontent.com/17569679/195963501-3b10c23a-4d68-432c-a4a6-82c671f87298.png)

**Comment by abraunegg** [comment]
@wanageeska 
So there is a number of items here to unpack in your question.

> Am I correct in thinking that if I add this large directory ('LIBRARY') to skip_dir in my config then it won't be downloaded when I --resync?

Correct - if the file already exists locally, it will not be downloaded, but the database is out-of-sync, thus, the local cache DB should be just updated with the file.

However - if you have 'LIBRARY' in your 'skip_dir' configuration, and are forced to issue a `--resync` - the application *should* be skipping that download and notifying you that this is the case - regardless of if you are using `--dry-run` or not.

From your screenshot however - it appears that this is not happening - which is odd ... so the question that needs to be asked is - what application version are you using: `onedrive --version`

**Comment by wanageeska** [comment, marked as answer]
@abraunegg 
To answer your question, I was running v2.4.21-1 from the AUR (just upgraded to v2.4.21-2 though).

However, I managed to figure out the issue while I was re-reading the `sync_dir` documentation, specifically this part:
> **Important:** Entries under `skip_dir` are relative to your `sync_dir` path.

D'oh! I was using the absolute path for the directory (i.e. `skip_dir="~/OneDrive/LIBARY"`). I changed it to `skip_dir="LIBRARY"` and when I ran `--resync` the client did indeed skip the directory.

Thanks for following it up anyway! And thanks for the excellent package!
